File: Reservation/src/main/java/net/travel/reservation/entites/Transaction.java
package net.travel.reservation.entites;


import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;


@Entity
@Table(name = "transactions")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Transaction {


    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long transactionId;




    // Exemple : "Acompte - ABC Consulting"
    @Column(nullable = false)
    private String libelle;



    private String description;



    @Column(nullable = false)
    private Double montant;



    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private TypeTransaction type;



    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private StatutTransaction statut;



    @Enumerated(EnumType.STRING)
    private ModePaiement modePaiement;



    private LocalDateTime dateTransaction;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(
            name = "reservation_id",
            nullable = true
    )
    private Reservation reservation;

}

File: reservation-frontend/src/app/core/models/transaction.ts
import { Espace } from "./espace/espace";
import { MethodePaiement } from "./paiement/MethodePaiement.enum";
import { Reservation } from "./reservations/reservation";
import { StatutTransaction } from "./StatutTransaction.enum";
import { TypeTransaction } from "./TypeTransaction.enum";


export class Transaction {
  transactionId?: number;
  libelle!: string;
  description?: string;
  montant!: number;
  type!: TypeTransaction;
  statut!: StatutTransaction;
  modePaiement?: MethodePaiement;
  dateTransaction?: string | Date;
  reservation?: Reservation | { reservationId: number };
  espace?: Espace | { espaceId: number };

  constructor(init?: Partial<Transaction>) {
    Object.assign(this, init);
  }
}